### PR TITLE
Sf fuels refactor

### PR DESCRIPTION
**Pull request by huangmy**
### Description:
First attempt to merge SF_fuels_refactor to NGEET_fates master (huangmy's fork)

### Collaborators:
@jkshuman
@rgknox
@ckoven
@rosiealice

### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [ ] If answers were expected to change, evaluation was performed and provided

### Test Results:

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

**Comment by huangmy** [comment]
@rgknox and @ckoven, I forked the NGEET/fates master and create a pull request from Jackie's branch at https://github.com/huangmy/fates/pull/1. There are quite a few conflicts there. Could you please take a look and let me know if it is OK to merge?

File: parameter_files/fates_params_14pfts.cdl [hunk 1]
```diff
@@ -1171,23 +1171,23 @@ data:
  fates_z0mr = 0.055, 0.055, 0.055, 0.055, 0.055, 0.055, 0.055, 0.055, 0.055, 
     0.055, 0.055, 0.055, 0.055, 0.055 ;
 
- fates_FBD = 4, 15.4, 16.8, 19.6, 999, 4 ;
+ fates_FBD = 15.4, 16.8, 19.6, 999, 4, 4 ;
```
Comment (rgknox) [inline parameter_files/fates_params_14pfts.cdl]: I'm a little confused @huangmy and @jkshuman, it seems the order of litter classes in both the base and the new, are a different order than what we currently have in master:

https://github.com/NGEET/fates/blob/master/parameter_files/fates_params_default.cdl#L1124

Does anyone remember when that change happened?

**Comment by jkshuman** [comment]
This change is a product of the refactor in Spitfire. I implemented that in
this branch. So yes, the values will need to be reordered in master.

------------------------------------------------------------------------
Jacquelyn Shuman, PhD
Project Scientist
Terrestrial Sciences Section
National Center for Atmospheric Research
PO Box 3000
Boulder, Colorado 80307-3000
USA

jkshuman@ucar.edu
office: +1-303-497-1787

File: parameter_files/fates_params_14pfts.cdl (hunk 1 above)
Comment (rgknox) [reply parameter_files/fates_params_14pfts.cdl]: well.. at least I thought that was the case.. But after reviewing the differences, the parameters in master, seem to match the base here.  Nevermind.

Comment (huangmy) [reply parameter_files/fates_params_14pfts.cdl]: So shall I go ahead to merge and we then worry about reordering the parameter files when merging to master?

**Comment by jkshuman** [comment]
I recommend accepting the branch parameter value order. This is correct
based on the refactor in the branch.

------------------------------------------------------------------------
Jacquelyn Shuman, PhD
Project Scientist
Terrestrial Sciences Section
National Center for Atmospheric Research
PO Box 3000
Boulder, Colorado 80307-3000
USA

jkshuman@ucar.edu
office: +1-303-497-1787

File: parameter_files/fates_params_14pfts.cdl (hunk 1 above)
Comment (huangmy) [reply parameter_files/fates_params_14pfts.cdl]: @jkshuman, I see, these are expected changes then. @rgknox, please let us know what to do.

**Comment by rgknox** [comment]
great, I see no issues either.  However, there are conflicts that need to be resolved.

for the three parameter files other than: parameter_files/fates_params_default.cdl

you can just do a git remove, ie:  git rm parameter_files/fates_params_14pfts.cdl

As for the default file, if the conflicts are very complicated, perhaps checkout the file from master into this branch instead of trying to merge and then resolve.

http://nicolasgallagher.com/git-checkout-specific-files-from-another-branch/

**Comment by rgknox** [comment]
so for the last command, (assuming your master branch is up to date)

git checkout master -- parameter_files/fates_params_default.cdl

and then apply the new parameters, again, manually.

**Comment by huangmy** [comment]
@rgknox @jkshuman I have merged @jkshuman's changes using command lines. Note sure if  parameter_files/fates_params_default.cdl is updated correctly or not though. Could you please take a look again?

**Pull request merged**